src/main/java/io/cdap/plugin/servicenow/connector/ServiceNowConnector.java

```diff
@@ -0,0 +1,167 @@
+  private static final String OBJECT_TABLE_LIST = "sys_db_object";
+  private static final String ENTITY_TYPE_TABLE = "table";
+  private static final String LABEL_NAME = "label";
+  private static final String OAUTH_URL_TEMPLATE = "%s/oauth_token.do";
+
+  private final ServiceNowConnectorConfig config;
+
+  ServiceNowConnector(ServiceNowConnectorConfig config) {
+    this.config = config;
+
+  }
+
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateCredentials(collector);
+  }
+
+  @Override
+  public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest browseRequest) throws IOException {
+    int count = 0;
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateCredentials(collector);
+    collector.getOrThrowException();
+    BrowseDetail.Builder browseDetailBuilder = BrowseDetail.builder();
+    TableList tables = listTables();
+    Table[] table = tables.getResult();
+    for (int i = 0; i < table.length; i++) {
+      String name = table[i].getName();
+      String label = table[i].getLabel();
+      BrowseEntity.Builder entity = (BrowseEntity.builder(name, name, ENTITY_TYPE_TABLE).
+        canBrowse(false).canSample(true));
+      entity.addProperty(LABEL_NAME, BrowseEntityPropertyValue.builder(label, BrowseEntityPropertyValue.
+        PropertyType.STRING).build());
+      browseDetailBuilder.addEntity(entity.build());
+      count++;
+    }
+    return browseDetailBuilder.setTotalCount(count).build();
+  }
+
+  public TableList listTables() {
+    ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+      config.getRestApiEndpoint(), OBJECT_TABLE_LIST);
+    String accessToken = null;
+    ServiceNowTableAPIClientImpl serviceNowTableAPIClient = new ServiceNowTableAPIClientImpl(config);
+    try {
+      accessToken = serviceNowTableAPIClient.getAccessToken();
+    } catch (OAuthSystemException | OAuthProblemException e) {
+      e.printStackTrace();
+    }
+    requestBuilder.setAuthHeader(accessToken);
+    requestBuilder.setAcceptHeader(MediaType.APPLICATION_JSON);
+    requestBuilder.setContentTypeHeader(MediaType.APPLICATION_JSON);
+    RestAPIResponse apiResponse = null;
+    apiResponse = serviceNowTableAPIClient.executeGet(requestBuilder.build());
+    if (!apiResponse.isSuccess()) {
+      LOG.error("Error - {}", getErrorMessage(apiResponse.getResponseBody()));
+      throw new RuntimeException();
+    } else {
+      String response = null;
+      response = apiResponse.getResponseBody();
+      Gson gson = new Gson();
+      TableList tables = gson.fromJson(response, TableList.class);
+      return tables;
+    }
+  }
+
+  @Override
+  public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSpecRequest connectorSpecRequest)
+    throws IOException {
```

Comment:
When does this method throw `IOException`?

Reply:
Removed